### ISSUE TITLE
Set a convention for mutation error handling

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#6249]
We need to establish a pattern for error handling and implement it where we're already using the `useMutation` hook in Settings.

See [the UnitRowSecondaryEmail](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/components/UnitRowSecondaryEmail/index.tsx) component and the "TODO" listed above the `useMutation` reference:

```java
  // TODO: DRY this up and don't import `sentryMetrics` into every component
  // needing a mutation - we must have an `onError` option in mutations
  // to allow tests to pass but providing one prevents an error from actually
  // throwing so we need to manually report it to Sentry.
  // See https://github.com/apollographql/react-apollo/issues/2614
```

Without this `onError` catch, or wrapping the mutation call in a `try`/`catch`, tests fail, and we want to avoid the need to manually set this on every mutation and avoid pulling in `sentryMetrics` into every component using a mutation.

Also see [the UnitRowRecoveryKey](https://github.com/mozilla/fxa/blob/main/packages/fxa-settings/src/components/UnitRowRecoveryKey/index.tsx) and a short discussion in [https://github.com/mozilla/fxa/pull/6224/files#r474080213](https://github.com/mozilla/fxa/pull/6224/files#r474080213). Do we want to show all GraphQL errors (messaging from the server like "invalid code"), then show a generic message like, "There was a problem with your request. Please try again" for network etc. errors?

We _probably_ don't want to send GQL errors to Sentry, but send other errors. Part of this is confirming this, and working with @MeridelW on the "general" error messaging. If she would like a more specific message per component (such as, "There was a problem removing your recovery key. Please try again."), it's probably best to do that in a follow up towards the end of this project.

- - -
TLDR;

* Confirm we want to show GQL error messages to the user and not send them to Sentry, and for other errors show a general error message and send the error to Sentry
* Make our `useMutation` error handling consistent according to the above (should we have a helper to avoid importing `sentryMetrics` everywhere?), and document the preferred approach in the `fxa-settings` README
* Grab the "general error" text from Meridel and display it where we should. If Meridel would like more specific errors, file a follow up issue to do and review at a later time.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2450)
┆Issue Number: FXA-2450
